scripts/validate_docstrings.py

```diff
@@ -491,8 +491,34 @@ def validate_one(func_name):
             errs.append('\t{}'.format(param_err))
 
     if doc.is_function_or_method:
-        if not doc.returns and "return" in doc.method_source:
-            errs.append('No Returns section found')
+        if not doc.returns:
+            if "return" in doc.method_source:
+                errs.append('No Returns section found')
+        else:
+            returns_errs = []
+            if len(doc.returns) == 1 and doc.returns[0][1]:
+                returns_errs.append('No name is to be provided when '
+                                    'returning a single value.')
+            for name, type_, desc in doc.returns:
+                desc = ''.join(desc)
+                name = '"' + name + '" ' if type_ else ''
+                if not desc:
+                    returns_errs.append('Return value {}has no '
```

Comment:
Hmm so is the assumption here that `name` always has a trailing whitespace? If so that seems rather fragile and should probably be fixed directly to remove that

Reply:
there should not be any trailing space in `name`.
I recognize that what's done here is not very clean :
* if `type_` is empty, which mainly occurs in fact when only one value is returned, which means in fact that `name` contains the type, then I don't want to print it in the error message.
* if `type_` is not empty, which means both name and type were provided, then I add a trailing space to `name`, for the error message to be correctly printed.

actually, the main problem I see here is that in the case of multiple returned values, if one of them has no name or type, the error message will not specify where the problem is :
with something like

```
Returns
-------
foo : int  
   The first value  
bar  
   The second value
```

will cause the following errors :
```
   Return value "foo" description should finish with "."
   Return value description should finish with "."
```

Reply:
I would be fine with a more generic error message then that doesn't use the name if it simplifies the code. The instances where multiple values are returned is few and far between to begin with so it shouldn't be hard as an end user to recognize where the issue appears if it comes up

Reply:
in b09c322e90e1b7ce7e5a00f6ef391ac48a08adac, I've removed value names from the error messages (and prevented from printing duplicate error messages)... but I'm not sure wether this simplifies the code...